### PR TITLE
Connection Handler and Event Handling Structure

### PR DESCRIPTION
**Pull request by jaybuurdd**
BACKEND
- Added websocket registration to our configuration process, now all event handler files added to /events folder will be registered at app run time.
- Added `connect` event handler (disconnect can go in here too). 

FRONTEND
- Added socket connection to our `vitetest` testcases, it will run a test check when you run `npm run test`. 
- To test socket connection live, I added a `TestSocketConnection.tsx` file to our `App.tsx`. Make sure backend server is running (`py app.py`) and then run client-side `npm run dev` if you check the logs from the server you'll see the user connected message. You can also check console in browser for the `Successfully connected to server!` message.

### CODE REVIEW

**Review by MrBisquit — APPROVED** [review]
Looks good.

File: frontend/src/tests/events/connection-event.test.tsx
Comment (believemanasseh) [inline frontend/src/tests/events/connection-event.test.tsx:1]: Isn't it better to maintain a consistent file name casing?

File: frontend/src/tests/events/socketLogic.js
Comment (believemanasseh) [inline frontend/src/tests/events/socketLogic.js:1]: Also, any particular reason this is .js?

File: frontend/src/tests/events/connection-event.test.tsx
Comment (jaybuurdd) [reply frontend/src/tests/events/connection-event.test.tsx:1]: I'm following the name casing that's been set for test cases to keep it consistent.

image

Comment (jaybuurdd) [reply frontend/src/tests/events/connection-event.test.tsx:1]: modified the other files to match the casing of this file.

**Pull request merged**